### PR TITLE
Ability to add validation to input fields

### PR DESCRIPTION
**Pull request by worksofliam**
### Changes

* Adds `minlength` and `maxlength` to input fields.
* Add real time validation to input fields to indicate that some fields are required
* Button fields can indicate that for them to be clicked all fields need to be valid
* Enter on a field doesn't work unless all fields are valid

### Checklist

* [x] have tested my change
* [ ] updated relevant documentation
* [ ] Remove any/all `console.log`s I added
* [ ] eslint is not complaining
* [ ] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/halcyon-tech/vscode-ibmi/blob/master/CONTRIBUTING.md)
* [ ] **for feature PRs**: PR only includes one feature enhancement.

### CODE REVIEW

**Comment by worksofliam** [comment]
I did consider adding `regexTest` as a validation option also, but wanted a second opinion on that idea before I did it.

File: src/api/CustomUI.ts [hunk 1]
```diff
@@ -293,23 +326,33 @@ export class CustomUI extends Section {
...
             }
 
-            console.log(groupButtons);
             // Now many buttons can be pressed to submit
-            for (const field of groupButtons) {
-                console.log('group button', field, document.getElementById(field));
+            for (const fieldData of groupButtons) {
+                const field = fieldData.id;
+                
+                console.log('group button', fieldData, document.getElementById(field));
```
Comment (edmundreinhardt) [inline src/api/CustomUI.ts:328]: do you want to keep the console.log

File: src/api/CustomUI.ts [hunk 2]
```diff
@@ -51,7 +52,7 @@ export class Section {
     return this;
   }
 
-  addInput(id: string, label: string, description?: string, options?: { default?: string, readonly?: boolean, rows?: number }) {
+  addInput(id: string, label: string, description?: string, options?: { default?: string, readonly?: boolean, rows?: number, minlength?: number, maxlength?: number }) {
```
Comment (edmundreinhardt) [inline src/api/CustomUI.ts:55]: cool and complicated
thanks for showing how to do the CustomUI by updating the HTML DOM

File: src/api/CustomUI.ts [hunk 3]
```diff
@@ -293,23 +326,33 @@ export class CustomUI extends Section {
...
-                button.onclick = (event) => {
-                    doDone(event, field);
-                };
-                button.onKeyDown = (event) => {
-                    doDone(event, field);
-                };
+
+                const submitButtonAction = (event) => {
+                  const isValid = fieldData.requiresValidation ? validateInputs() : true;
+                  console.log({requiresValidation: fieldData.requiresValidation, isValid});
```
Comment (edmundreinhardt) [inline src/api/CustomUI.ts:333]: another console.log

**Review by edmundreinhardt — COMMENTED** [review]
Nice piece of work!

File: src/api/CustomUI.ts (hunk 1 above)
Comment (worksofliam) [reply src/api/CustomUI.ts:328]: Yes, we've needed this when debugging in production before. Safe to keep it for now.

**Comment by worksofliam** [comment]
Added regex test support and also removed some console logs.

**Comment by Wright4i** [comment]
@worksofliam  - Other than those 3 comments above I approve! I tried thoroughly to break this change and it seems all the fields are validating correctly. Great user experience change and I hope this reduces some issues for newcomers. Awesome work.

**Pull request merged**